test/unit/test/pretty_printing_test.cpp

```diff
@@ -44,18 +44,27 @@ TEST(pretty_printing, gtest_output)
 
     EXPECT_EQ(gtest_str(std::make_tuple<int, int>(42, -10)), "(42, -10)"s);
     EXPECT_EQ(debug_str(std::make_tuple<int, int>(42, -10)), "(42,-10)"s);
-}
-
-TEST(pretty_printing, std_output)
-{
-    EXPECT_EQ(gtest_str(std::vector<std::vector<int>>{{0,1}, {2,3}, {1,2}, {0}}), "[[0,1],[2,3],[1,2],[0]]"s);
-    EXPECT_EQ(debug_str(std::vector<std::vector<int>>{{0,1}, {2,3}, {1,2}, {0}}), "[[0,1],[2,3],[1,2],[0]]"s);
 
+#if GTEST_INTERNAL_HAS_VARIANT // googletest >1.10.0
+    EXPECT_EQ(gtest_str(std::variant<int>{0}), "('int(index = 0)' with value 0)"s);
+#else // googletest <= 1.10.0
     EXPECT_EQ(gtest_str(std::variant<int>{0}), "0"s);
+#endif
     EXPECT_EQ(debug_str(std::variant<int>{0}), "0"s);
 
+#if GTEST_INTERNAL_HAS_OPTIONAL // googletest >1.10.0
+    EXPECT_EQ(gtest_str(std::optional<int>{}), "(nullopt)"s);
+#else // googletest <= 1.10.0
     EXPECT_EQ(gtest_str(std::optional<int>{}), "<VALUELESS_OPTIONAL>"s);
+#endif
     EXPECT_EQ(debug_str(std::optional<int>{}), "<VALUELESS_OPTIONAL>"s);
+}
+
+TEST(pretty_printing, std_output)
+{
+    // if any of these tests get supported by googletest move them to gtest_output
```

Comment:
Is there a common understanding of what it means that these test are supported by googletest?

To me it seems like it is already support. It seems like I am missing something.

Reply:
> To me it seems like it is already support. It seems like I am missing something.

There are indirectly supported through our debug_stream (hacky hack). That's why the output for debug stream and google test look alike. But it could be that gtest will handle the output on it's own at some point and then it whould be moved.